src/rashdf/geom.py

```diff
@@ -106,10 +106,15 @@ def mesh_areas(self) -> GeoDataFrame:
         mesh_area_names = self.mesh_area_names()
         if not mesh_area_names:
             return GeoDataFrame()
-        mesh_area_polygons = [
-            Polygon(self[f"{self.FLOW_AREA_2D_PATH}/{n}/Perimeter"][()])
-            for n in mesh_area_names
-        ]
+
+        mesh_area_polygons = []
+        for n in mesh_area_names:
+            try:
+                mesh_area_polygons.append(
+                    Polygon(self[f"{self.FLOW_AREA_2D_PATH}/{n}/Perimeter"][()])
+                )
+            except Exception:
+                raise RasGeomHdfError(f"Mesh {n} not found.")
```

Comment:
Catch `KeyError` since we know that's the specific error we want to handle rather than the too-broad `Exception` (also, quotes around the mesh name for readability):
```python
            try:
                mesh_area_polygons.append(
                    Polygon(self[f"{self.FLOW_AREA_2D_PATH}/{n}/Perimeter"][()])
                )
            except KeyError as e:
                raise RasGeomHdfError(f"Data for mesh '{n}' not found.") from e
```